triplet/main.py

```diff
@@ -151,7 +145,13 @@ def convert_and_transform(img, transform):
         return img_a, img_p, img_n
 
     def __len__(self):
-        return self.triplets.shape[0]
+        try:
+            if self.train:
+                return self.triplets.shape[0]
+            else:
+                return self.matches.shape[0]
+        except:
+            pass
 
```

Comment:
No I'm sorry, but I don't think I understand the problem. This `__len__` will never be called by the parent class, what's the error you're getting? I think it should be safe to remove the `try` block